R/TealAppDriver.R

```diff
@@ -311,39 +311,70 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
       active_filters[[dataset_name]]
     },
     #' @description
-    #' Add a new variable from the dataset to be filtered.
+    #' Adds a new filter variable to the dataset. Supports both standard datasets and MAE datasets.
     #'
     #' @param dataset_name (character) The name of the dataset to add the filter variable to.
+    #' @param sample_name (character) The name of the sample/experiment.
     #' @param var_name (character) The name of the variable to add to the filter panel.
+    #' @param filter_type (character) Specifies the type of filter. Default is "data.frame".
     #' @param ... Additional arguments to be passed to `shinytest2::AppDriver$set_inputs`
     #'
     #' @return The `TealAppDriver` object invisibly.
-    add_filter_var = function(dataset_name, var_name, ...) {
+    add_filter_var = function(dataset_name, sample_name, var_name, filter_type = "data.frame", ...) {
       checkmate::check_string(dataset_name)
       checkmate::check_string(var_name)
-      self$set_input(
-        sprintf(
+      checkmate::assert_choice(filter_type, c("data.frame", "subject", "gene", "sample"))
+
+      input_id <- switch(filter_type,
+        `data.frame` = sprintf(
           "%s-add-%s-filter-var_to_add",
           self$active_filters_ns(),
           dataset_name
         ),
-        var_name,
-        ...
+        subject = sprintf(
+          "%s-add-%s-subjects-var_to_add",
+          self$active_filters_ns(),
+          dataset_name
+        ),
+        gene = {
+          checkmate::check_string(sample_name)
+          sprintf(
+            "%s-add-%s-%s-row_to_add",
+            self$active_filters_ns(),
+            dataset_name,
+            sample_name
+          )
+        },
+        sample = {
+          checkmate::check_string(sample_name)
+          sprintf(
+            "%s-add-%s-%s-col_to_add",
+            self$active_filters_ns(),
+            dataset_name,
+            sample_name
+          )
+        }
       )
+
+      self$set_input(input_id, var_name, ...)
       invisible(self)
     },
     #' @description
     #' Remove an active filter variable of a dataset from the active filter variables panel.
     #'
     #' @param dataset_name (character) The name of the dataset to remove the filter variable from.
     #' If `NULL`, all the filter variables will be removed.
+    #' @param sample_name (character) The name of the sample, required for "gene" and "sample" filter types.
     #' @param var_name (character) The name of the variable to remove from the filter panel.
     #' If `NULL`, all the filter variables of the dataset will be removed.
+    #' @param filter_type (character) Specifies the type of filter: "data.frame", "subject", "gene", or "sample".
```

Comment:
Same comment as above. As we have `sample_name` as `NULL` so it makes more sense for me when `filter_type` is also `NULL`. Perhaps it can be called something else more specific like `mae_entity`/`mae_element`/`mae_component`.

Reply:
okay in this case keeping it null make sense as we won't have `data.frame` only c("subject", "gene", "sample"). I am okay with this . I will make changes anyway because the filter type was a bit vague.